### PR TITLE
Add a GA workflow to automatically add labels to PRs

### PR DESCRIPTION
**Pull request by kmshin1397**
Check modified files in PRs and automatically add the following labels where relevant (some labels were already in the repository, some I just created):
- config-change
- demo-data
- dependencies
- docker
- documentation
- formatting-and-linting
- javascript
- migration
- needs-migration?
- python
- workflows

See inline comments for detailed specification of what triggers each label. Feel free to suggest more labels if you see come up with any.

Addresses #1707

### CODE REVIEW

**Comment by kmshin1397** [comment]
PR on my fork where I tested this, if you're interested: https://github.com/kmshin1397/skyportal/pull/3

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  - "static/js/**/*.jsx"
+
+# Add 'migration' label to any change within the 'alembic' directory
+migration:
+  - alembic/versions/*
+
+# Add 'needs-migration?' label to any change to models.py without changes to
+# the Alembic versions
+needs-migration?:
+  - any: ["skyportal/models.py"]
```
Comment (acrellin) [inline .github/labeler.yml:51]: any way to see if the pinned baselayer/app/models.py was touched, too?

Comment (kmshin1397) [reply .github/labeler.yml:51]: Ah yeah good point - I'll look into it

**Comment by stefanv** [comment]
This is cute, thanks!

We should think about which labels are helpful in development.  I think, e.g., that `python` and `javascript` are not.  Config changes and migrations clearly are important.  How about the others?  What is the motivation for each?

**Comment by kmshin1397** [comment]
@stefanv Here are my thoughts on the other labels. I'm definitely open to discussion here:

- demo-data
  - Developers may see that a `demo-data` labelled PR has recently been merged, and to rerun `make load_demo_data` so that their development environment has test data in line with what is on master. 
- dependencies
  - The dependabot PRs already use this label automatically, I believe. So I added this to match that behavior. I imagine it's good for reviewers to immediately see when new dependencies are being added/updated.
- docker
  - I imagine if there are developers using Dockerized deployments locally for development, they may want to quickly see if any recent PRs include changes to those configurations and decide if they want to rebuild local images? I was also thinking it would be good to emphasize if there are Docker/Docker Compose changes to nudge reviewers to actually try building/running them but I guess that's less of a reason once #1687 is merged. 
- documentation
  - No profound reason - this label already existed (probably to tag issues) so I decided to support it since it's already there
- formatting-and-linting
  - If we ever decide to change certain style rules/guidelines and modify linting/formatting rules accordingly it would be good to highlight that PR so developers can weigh in and be aware of those changes, as all developers are expected to adhere to these style rules.
- javascript
  - This was also something that was already there so I just included it. It seems like it might just be a label used by dependabot? 
- needs-migration?
  - Although #1680 will error if migrations are actually missing, this tag will provide a very clear human-readable message to the developer that they may need to add a migration script without having to go into the GA run and see what actually threw the red X. To be fair, this will also include false positives in that any changes (even innocuous ones like comments) to `models.py` without a subsequent migration will trigger the label. It's why I have the question mark, but we could also just forgo this label.
- python
  - Same thing with the `javascript` tag 
- workflows
  - Honestly just added as an afterthought for completeness as it was the only top-level directory in the repository that didn't really have a label associated with it.

**Comment by stefanv** [comment]
Thanks for that thorough justification!  I'd remove the javascript and Python tags, then we see how it goes with the rest.

**Comment by kmshin1397** [comment]
> Thanks for that thorough justification! I'd remove the javascript and Python tags, then we see how it goes with the rest.

Sounds good!

File: .github/labeler.yml (hunk 1 above)
Comment (kmshin1397) [reply .github/labeler.yml:46]: @acrellin Hmm I don't think this is straightforward to do. I enabled the debug logs for this action, and the only thing it gets back from the github API for files changed is `baselayer` and there is no info of what changed in the submodule.

**Comment by acrellin** [comment]
@kmshin1397 in that case, can we add `needs-migration?` whenever baselayer changes so that we can make sure there are no changes there to DB models? Then it can be manually removed after someone has checked

**Comment by kmshin1397** [comment]
@acrellin Sure, that can be done

**Comment by kmshin1397** [comment]
@stefanv I've removed the javascript and python tags here. Are you good with this going in?

**Comment by stefanv** [comment]
Thanks!

**Pull request merged**